src/Servers/Kestrel/Core/src/Internal/ConfigurationReader.cs

```diff
@@ -97,7 +98,8 @@ private IEnumerable<EndpointConfig> ReadEndpoints()
                     ConfigSection = endpointConfig,
                     Certificate = new CertificateConfig(endpointConfig.GetSection(CertificateKey)),
                     SslProtocols = ParseSslProcotols(endpointConfig.GetSection(SslProtocolsKey)),
-                    ClientCertificateMode = ParseClientCertificateMode(endpointConfig[ClientCertificateModeKey])
+                    ClientCertificateMode = ParseClientCertificateMode(endpointConfig[ClientCertificateModeKey]),
+                    SNI = ReadSni(endpointConfig.GetSection(SNIKey)),
```

Comment:
Consider a global SNI section as well that takes priority over the default cert, but is overridden by endpoint certs.